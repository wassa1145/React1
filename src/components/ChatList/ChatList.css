.chats-link {
	border:  1px solid #999;
	height: 48px;
	display: block;
	flex-grow: 1;
}

.chat-item {
	margin-bottom: 8px;
}

.chat-list-wrapper {
	height: 100%;
	max-width: 300px;
	display: flex;
	flex-direction: column;
	justify-content: space-between;
}
.form-input {
	width: 100%;
	margin-bottom: 15px !important;
}

.chats-link_active {
	border-right: 4px solid #363333;
}
.row {
	display: flex;
	justify-content: space-between;
}

.button-delete {
	min-width: 20px !important;
	color: #363333 !important;
}
