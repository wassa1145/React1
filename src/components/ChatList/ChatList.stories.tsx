import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { ChatList } from './ChatList';
import { BrowserRouter } from 'react-router-dom';

export default {
  title: 'MyComponents/ChatList',
  component: ChatList,
  argTypes: {
    onAddChat: { action: 'click' },
    deleteChat: { action: 'click' },
  },
} as ComponentMeta<typeof ChatList>;

const Template: ComponentStory<typeof ChatList> = (args) => (
  <BrowserRouter>
    <ChatList {...args} />
  </BrowserRouter>
);

export const Primary = Template.bind({});
Primary.args = {
  chats: [
    { id: '1', name: 'First' },
    { id: '2', name: 'Second' },
    { id: '3', name: 'Third' },
  ],
};
