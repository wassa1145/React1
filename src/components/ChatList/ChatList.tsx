import React, { FC, useState } from 'react';
import { NavLink } from 'react-router-dom';
import { List as ListUI, TextField } from '@mui/material';
import { Button as ButtonUI } from '@mui/material';
import ListItem from '@mui/material/ListItem';
import ListItemButton from '@mui/material/ListItemButton';
import ListItemText from '@mui/material/ListItemText';
import './ChatList.css';

import { Button } from '../FormFunc/components/Button/Button';
import { useDispatch, useSelector } from 'react-redux';
import { selectChatList } from '../../store/chats/selectors';
import { addChat, deleteChat } from '../../store/chats/actions';

export const ChatList: FC = () => {
  const [name, setName] = useState('');

  const dispatch = useDispatch();

  const chatList = useSelector(
    selectChatList,
    (prev, next) => prev.length === next.length
  );

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    if (name) {
      dispatch(addChat(name));
      setName('');
    }
  };

  return (
    <div className="chat-list-wrapper" data-testid="chat-list">
      <div>
        <h1>Список чатов</h1>
        <ListUI>
          {chatList.map((chat) => (
            <ListItem
              key={chat.id}
              role="chat"
              className="chat-item"
              secondaryAction={
                <ButtonUI
                  onClick={() => dispatch(deleteChat(chat.name))}
                  className="button-delete"
                  role="button-delete"
                >
                  X
                </ButtonUI>
              }
              disablePadding
            >
              <NavLink
                to={`/chats/${chat.name}`}
                className={({ isActive }) =>
                  isActive ? 'chats-link chats-link_active' : 'chats-link'
                }
              >
                <ListItemButton>
                  <ListItemText primary={chat.name} />
                </ListItemButton>
              </NavLink>
            </ListItem>
          ))}
        </ListUI>
      </div>
      <form onSubmit={handleSubmit}>
        <TextField
          variant="outlined"
          className="form-input"
          placeholder="Введите название чата"
          role="input-chat-name"
          value={name}
          onChange={(e) => setName(e.target.value)}
        />
        <Button name="Добавить новый чат" disabled={!name} />
      </form>
    </div>
  );
};
