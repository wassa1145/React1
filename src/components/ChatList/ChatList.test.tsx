import React from 'react';
import { ChatList } from './ChatList';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('ChatList', () => {
  const chats = [
    { id: '1', name: 'First' },
    { id: '2', name: 'Second' },
  ];

  beforeEach(() => {
    render(<ChatList chats={chats} />);
  });

  it('render component', () => {
    expect(screen.getByTestId('chat-list')).toBeInTheDocument();
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<ChatList chats={chats} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render component with text', () => {
    expect(screen.getByText(/Second/)).toBeInTheDocument();
  });
});
