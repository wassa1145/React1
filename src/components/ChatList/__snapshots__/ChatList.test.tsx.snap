// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatList render with snapshot 1`] = `
<DocumentFragment>
  <div
    class="chat-list-wrapper"
    data-testid="chat-list"
  >
    <div>
      <h1>
        Список чатов
      </h1>
      <ul
        class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters chat-item css-g28sfz-MuiListItem-root"
          role="chat"
        >
          <a
            class="chats-link"
            href="/chats/First"
          >
            <div
              class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  First
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </a>
          <div
            class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root button-delete css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              role="button-delete"
              tabindex="0"
              type="button"
            >
              X
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters chat-item css-g28sfz-MuiListItem-root"
          role="chat"
        >
          <a
            class="chats-link"
            href="/chats/Second"
          >
            <div
              class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Second
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </a>
          <div
            class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root button-delete css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              role="button-delete"
              tabindex="0"
              type="button"
            >
              X
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </li>
      </ul>
    </div>
    <form>
      <div
        class="MuiFormControl-root MuiTextField-root form-input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        role="input-chat-name"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r2:"
            placeholder="Введите название чата"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled form__button css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        data-testid="submit-button"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        Добавить новый чат
      </button>
    </form>
  </div>
</DocumentFragment>
`;
